dao/src/main/java/se/fortnox/reactivewizard/db/statement/StatementDebug.java

```diff
@@ -0,0 +1,27 @@
+package se.fortnox.reactivewizard.db.statement;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.PreparedStatement;
+
+public class StatementDebug {
+    private static final Logger LOG = LoggerFactory.getLogger(StatementDebug.class);
+    static boolean ENABLE_STATEMENT_DEBUG = Boolean.parseBoolean(System.getenv("dao.debug"));
```

Comment:
Can this be final so that the if-statement below cannot change at runtime and the JVM can "remove" the code?

Reply:
We have to be able to change the flag in the tests. Unsure if Mockito has a nice api for that. Perhaps the value of the field can be changed with reflection but I don't think that is an OK solution to this problem.  We could create a static method, getEnableStatementDebug, that returns the ENABLE_STATEMENT_DEBUG, which can then be mocked by enabling Mockito's inline mock maker. That would solve one part of the problem, not sure though if JVM will inline the log method calls when you call getEnableStatementDebug instead of just checking the ENABLE_STATEMENT_DEBUG field.

Reply:
Why do you need to change the flag in the tests? Can't you just decide that all tests are run with the flag on or off?

Reply:
I was thinking specifically of [these tests](https://github.com/FortnoxAB/reactive-wizard/pull/465/commits/28ec8be3a02506ac66dc3655b012fc049f98071a#diff-81166375d5db84e39cfb0e42b65f7c220291d8d7dab049420b4845c768cf10cb). I have manually tested these cases instead.

Reply:
Personally, I would rather have the statement final than having it possible to automaticall test the behaviour when it is off.